rclpy/rclpy/parameter_client.py

```diff
@@ -0,0 +1,310 @@
+        :return: ``Future`` with the result of the request.
+        """
+        request = SetParameters.Request()
+        request.parameters = parameters
+        future = self._set_parameter_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def describe_parameters(
+        self,
+        names: List[str],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Describe parameters given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.DescribeParameters.Response``.
+
+        :param names: List of parameter names to describe
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = DescribeParameters.Request()
+        request.names = names
+        future = self._describe_parameters_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def get_parameter_types(
+        self,
+        names: List[str],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Get parameter types given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.GetParameterTypes.Response``.
+
+        Parameter type definitions are given in rcl_interfaces.msg.ParameterType
+
+        :param names: List of parameter names to get types for.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = GetParameterTypes.Request()
+        request.names = names
+        future = self._get_parameter_types_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def set_parameters_atomically(
+        self,
+        parameters: Sequence[Parameter],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Set parameters atomically.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.SetParametersAtomically.Response``.
+
+        :param parameters: Sequence of parameters to set.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = SetParametersAtomically.Request()
+        request.parameters = parameters
+        future = self._set_parameters_atomically_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def delete_parameters(
+        self, names: List[str], callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Unset parameters with given names.
+
```

Comment:
It might be worth mentioning here that only parameters declared as dynamically typed can be unset.